web/docs/auth/auth-hooks.md

```diff
@@ -567,3 +722,132 @@ The hook receives an object as **input** with the following properties:
+
+```js title="src/auth/hooks.js"
+export const onBeforeLogin = async ({
+  providerId,
+  prisma,
+  req,
+}) => {
+  // Hook code goes here
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/auth/hooks.ts"
+import type { OnBeforeLoginHook } from 'wasp/server/auth' 
+
+export const onBeforeLogin: OnBeforeLoginHook = async ({
+  providerId,
+  prisma,
+  req,
+}) => {
+  // Hook code goes here
+}
+```
+</TabItem>
+</Tabs>
+
+The hook receives an object as **input** with the following properties:
+- [`providerId: ProviderId`](#providerid-fields)
+
+- `prisma: PrismaClient`
+
+  The Prisma client instance which you can use to query your database.
+
+- `req: Request`
+
+  The [Express request object](https://expressjs.com/en/api.html#req) from which you can access the request headers, cookies, etc.
+
+Wasp ignores this hook's **return value**.
+
+### The `onAfterLogin` hook
+
+<Tabs groupId="js-ts">
+
+<TabItem value="js" label="JavaScript">
+
+```js title="src/auth/hooks.js"
+export const onAfterLogin = async ({
+  providerId,
+  user,
+  oauth,
+  prisma,
+  req,
+}) => {
+  // Hook code goes here
+}
+```
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title="src/auth/hooks.ts"
+import type { OnAfterLoginHook } from 'wasp/server/auth'
+
+export const onAfterLogin: OnAfterLoginHook = async ({
+  providerId,
+  user,
+  oauth,
+  prisma,
+  req,
+}) => {
+  // Hook code goes here
+}
+```
+</TabItem>
+</Tabs>
+
+The hook receives an object as **input** with the following properties:
+- [`providerId: ProviderId`](#providerid-fields)
+
+- `user: User`
+
+  The user object of the user who logged in.
+
+- [`oauth?: OAuthFields`](#oauth-fields)
+
+- `prisma: PrismaClient`
+
+  The Prisma client instance which you can use to query your database.
+
+- `req: Request`
+
+  The [Express request object](https://expressjs.com/en/api.html#req) from which you can access the request headers, cookies, etc.
+
+Wasp ignores this hook's **return value**.
+
+### ProviderId fields
+
+The `providerId` object is passed to the `onBeforeSignup`, `onAfterSignup`, `onBeforeLogin`, and `onAfterLogin` hooks.
+
+It has the following fields:
+
+- `providerName: string`
+
+  The provider's name (e.g. `'email'`, `'google'`, `'github`)
+
+- `providerUserId: string`
+
+  The user's unique ID in the provider's system (e.g. email, Google ID, GitHub ID)
+
+### OAuth fields
+
+The `oauth` object is passed to the `onAfterSignup` and `onAfterLogin` hooks only when the user is authenticated with [Social Auth](./social-auth/overview.md).
+
+It has the following fields:
+
+- `accessToken: string`
+
+  The OAuth access token that can be used to make requests to the provider's API on the user's behalf.
+
+- `uniqueRequestId: string`
+
+  The unique request ID for the OAuth flow (you might know it as the `state` parameter in OAuth.)
+
+  You can use the unique request ID to get the data saved in the `onBeforeOAuthRedirect` hook.
```

Comment:
Unnecessary passive form (to get the data saved -> to save the data). I'd expect Grammarly to flag this